superchain/superchain.go

```diff
@@ -483,7 +483,18 @@ type GenesisAccount struct {
 	Nonce    uint64        `json:"nonce,omitempty"`
 }
 
+// OptimismConfig is the optimism config.
+type OptimismConfig struct {
+	EIP1559Elasticity        uint64  `json:"eip1559Elasticity"`
+	EIP1559Denominator       uint64  `json:"eip1559Denominator"`
+	EIP1559DenominatorCanyon *uint64 `json:"eip1559DenominatorCanyon,omitempty"`
+}
+type GenesisConfig struct {
+	// Optimism config, nil if not active
+	Optimism *OptimismConfig `json:"optimism,omitempty"`
+}
 type Genesis struct {
+	Config GenesisConfig `json:"config"`
```

Comment:
Why don't we directly add the `*OptimismConfig` as a field here? To better match json hydration?